src/Microsoft.Extensions.ApiDescription.Client/build/Microsoft.Extensions.ApiDescription.Client.targets

```diff
@@ -279,8 +295,17 @@
       unique.
     -->
     <ItemGroup>
-      <Compile Remove="@(ServiceFileReference -> '%(OutputPath)')" />
-      <Compile Include="@(ServiceFileReference -> '%(OutputPath)')" SourceDocument="%(FullPath)" />
+      <TypeScriptCompile Remove="@(ServiceFileReference -> '%(RelativeOutputPath)')" />
+      <TypeScriptCompile Include="@(ServiceFileReference -> '%(RelativeOutputPath)')"
+          Condition="'%(ServiceFileReference.Extension)' == '.ts'">
+        <SourceDocument>%(ServiceFileReference.FullPath)</SourceDocument>
+      </TypeScriptCompile>
+
+      <Compile Remove="@(ServiceFileReference -> '%(RelativeOutputPath)')" />
+      <Compile Include="@(ServiceFileReference -> '%(RelativeOutputPath)')"
+          Condition="'%(ServiceFileReference.Extension)' != '.ts'">
```

Comment:
Should this say `'%(ServiceFileReference.Extension)' == '$(Language)'` or `'%(ServiceFileReference.Extension)' == 'C#'`?

Reply:
Switched to use `$(DefaultLanguageSourceExtension)` and added real support for `%(OutputPath)` being a directory.